src/client/datascience/kernel-launcher/localPythonAndRelatedNonPythonKernelSpecFinder.ts

```diff
@@ -128,7 +128,8 @@ export class LocalPythonAndRelatedNonPythonKernelSpecFinder extends LocalKernelS
                     if (
                         kernelspec.language === PYTHON_LANGUAGE &&
                         hideDefaultKernelSpecs &&
-                        kernelspec.name.toLowerCase().match(isDefaultPythonKernelSpecName)
+                        (kernelspec.name.toLowerCase().match(isDefaultPythonKernelSpecName) ||
+                            kernelspec.display_name.toLowerCase() === 'python 3 (ipykernel)')
```

Comment:
Isn't that name a bug? What would cause it be called that?

Reply:
Oh forgot to mention, when installing ipykernel6 it creates a kernel spec by default in each python environment named `Python 3 (ipykernel)` & this causes issues going forward.
I.e. we need to hide this the same way we hide default `python3` kernels in interpreters.

@rchiodo /cc